File: src/main/java/by/devpav/serfor/services/impl/image/ImageResizerImpl.java
package by.devpav.serfor.services.impl.image;

import org.springframework.stereotype.Component;

import java.awt.*;
import java.awt.image.BufferedImage;

@Component
public class ImageResizerImpl implements ImageResizer {

    @Override
    public BufferedImage resize(BufferedImage img, int height, int width) {
        Image tmp = img.getScaledInstance(width, height, Image.SCALE_SMOOTH);
        BufferedImage resized = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
        Graphics2D g2d = resized.createGraphics();
        g2d.drawImage(tmp, 0, 0, null);
        g2d.dispose();
        return resized;
    }

}

File: src/main/kotlin/by/devpav/serfor/service/image/ImageResizerImpl.kt
package by.devpav.serfor.service.image

import by.devpav.serfor.services.impl.image.ImageResizer
import org.springframework.stereotype.Component
import java.awt.Image
import java.awt.image.BufferedImage

@Component
class ImageResizerImpl : ImageResizer {

    override fun resize(img: BufferedImage, height: Int, width: Int): BufferedImage {
        val tmp = img.getScaledInstance(width, height, Image.SCALE_SMOOTH)
        val resized = BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB)
        val g2d = resized.createGraphics()
        g2d.drawImage(tmp, 0, 0, null)
        g2d.dispose()
        return resized
    }

}
